libraries/esp8266/examples/SigmaDeltaDemo/SigmaDeltaDemo.ino

```diff
@@ -0,0 +1,84 @@
+/*
+  Sigma Delta Generator example
+  
+  This example demonstrates the use of the ESP8266 internal hardware sigma delta source.
+  Each GPIO pin or multiple pins can be connected to the one sigma delta source.
+  THE sigma delta output frequency is defined by the ESP8266 hardware as:
+
+  FREQ = 80,000,000/prescaler * target /256  HZ,          0<target<128
+  FREQ = 80,000,000/prescaler * (256-target) /256  HZ,    128<target<256
+  with :
+  target: 0-255, defines the 8-bit duty cycle of the output signal, , duty cycle = target/256
+  prescaler: 0-255, reduce the output frequencies
+  
+  so both the target and prescaler will affect the freq.
+  
+  Differences with Arduino library analogWrite :
+  - the arduino analogWrite pwm uses a hardware timer and ISR to generate the pwm signal. The sigma delta pwm doesn't need a hardware timer. 
+    The FRC1 timer can then be used for other purposes.
+  - the arduino analogWrite pwm produces a more or less fixed frequency around 900Hz. The sigma delta source produces a variable output frequency, 
+    that depends on the duty cycle. Example, with prescaler = 10, the output frequency varies between 31,kHz and 4MHz
+  - the sigma delta source produces much higher frequencies. This allows the output to be easily smoothed by a RC-filter and realise a decent 8-bit DAC
+  - the arduino analogWrite pwm has a higher resolution, default 12-bit, and up to 16-bit resolution. The sigma delta source has only 8-bit resolution
+    
+  The example doesn't require additional hardware.
+  
+  With an RC-filter (for example R=17k & C=100nF), a 'DC' output signal can be obtained with voltage between 0 and 3.3V
+  For higher value capacitors, it might be necessary to add a high-speed buffer between the esp8266 output and RC-filter to limit the current ac load on the 
+  pin output driver.
+
+*/
```

Comment:
Could you please add attribution and license (CC0 or PD) to the example header?

Reply:
Could you please add attribution and license (CC0 or PD) to the example header?

Reply:
PD i guess?